### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by cferreras** [DrexHD/AntiXray#53]
```
Description: Exception in server tick loop

java.lang.ClassCastException: class net.minecraft.class_2841$class_6563$Anonymous$1013a89a57d047ee976a6402ccb18f6d cannot be cast to class net.minecraft.class_6880 (net.minecraft.class_2841$class_6563$Anonymous$1013a89a57d047ee976a6402ccb18f6d and net.minecraft.class_6880 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @337d0578)
	at net.minecraft.class_2378$1.method_10206(class_2378.java:186)
	at net.minecraft.class_2834.method_12290(class_2834.java:100)
	at net.minecraft.class_2841$class_6561.method_38306(class_2841.java:321)
	at net.minecraft.class_2841.method_12327(class_2841.java:275)
	at net.minecraft.class_2826.method_12260(class_2826.java:176)
	at net.minecraft.class_6603.method_38589(class_6603.java:80)
	at net.minecraft.class_6603.<init>(class_6603.java:42)
	at net.minecraft.class_2672.mixinextras$bridge$newclass_6603$15(class_2672.java)
	at net.minecraft.class_2672.wrapOperation$zze000$antixray$setChunkPacketInfoArgument(class_2672.java:559)
	at net.minecraft.class_2672.<init>(class_2672.java:28)
	at net.minecraft.class_8608.method_52388(class_8608.java:88)
	at net.minecraft.class_8608.mixinextras$bridge$method_52388$13(class_8608.java)
	at net.minecraft.class_8608.wrapOperation$zzk000$antixray$setBatchStartPacketArgument(class_8608.java:556)
	at net.minecraft.class_8608.method_52386(class_8608.java:81)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:1045)
	at net.minecraft.class_3176.method_3813(class_3176.java:299)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:912)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:697)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:281)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

antixray-fabric-1.4.6+1.20.4.jar

**Comment by DrexHD** [comment]
Are you sure this is with version 1.4.6? Can you send the entire crash log?

**Comment by cferreras** [comment]
It is.
[crash-2024-08-17_01.37.43-server.txt](https://github.com/user-attachments/files/16645264/crash-2024-08-17_01.37.43-server.txt)

**Comment by DrexHD** [comment]
The crash report states `antixray: AntiXray 1.4.5`. I will close this as invalid for now, unless you can send a crash report with [1.4.6+1.21.1](https://modrinth.com/mod/anti-xray/version/1.4.6+1.21.1)

**Comment by cferreras** [comment]
I downloaded it again and seems to work.

**Comment by 0elek** [comment]
will 1.4.6 support 1.21?

edit i seems the fix(1.4.6) is done for 1.21 but not yet relesed

**Comment by 0elek** [comment]
Made a fork that reverts the `branch 1.21` to 1.21 and added a new branch for 1.21.1.
Building from there fixes the issue.

**Comment by DrexHD** [comment]
> edit i seems the fix(1.4.6) is done for 1.21 but not yet relesed

The fix has been applied to the 1.21(.1) branch and [released on modrinth](https://modrinth.com/mod/anti-xray/versions?g=1.21.1), but modrinth doesn't seem to feature it! 1.21.1 is a hotfix that shouldn't really break any mod and anyone running 1.21 should update to it!
